### PR TITLE
Headingblock

### PR DESCRIPTION
**Pull request by arjun87743**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix https://adobe-acs-group.atlassian.net/browse/WRLDBAEMS-122

Test URLs:
- Before: https://main--world-bank--aemsites.hlx.live/
- After: https://headingblock--world-bank--aemsites.hlx.live/en/drafts/arjun/headingblock

Authoring URL: https://author-p136806-e1403562.adobeaemcloud.com/ui#/@worldbank/aem/universal-editor/canvas/author-p136806-e1403562.adobeaemcloud.com/content/world-bank/global/en/drafts/arjun/headingblock.html?ref=headingblock

### CODE REVIEW

File: blocks/heading/heading.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+function updateHeadingStructure(block) {
+  const headingBlock = block;
+  if (!headingBlock) return;
+
+  const children = [...headingBlock.children];
+  if (children.length < 7) return;
```
Comment (thomsebastin) [inline blocks/heading/heading.js:6]: If optional fields are not authored, will this still be 7?

File: blocks/heading/heading.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    mainHeadingDiv,
+    headerTagDiv,
+    suffixDiv,
+    descriptionDiv,
+    bottomLineDiv,
+  ] = children;
+
+  const getTextContent = (div) => div?.querySelector('p')?.textContent || '';
+
+  const eyebrowText = getTextContent(eyebrowDiv);
```
Comment (thomsebastin) [inline blocks/heading/heading.js:19]: textcontent might have whitespaces. Always good to trim.

File: blocks/heading/heading.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  const descriptionText = getTextContent(descriptionDiv);
+  const addBottomLine = getTextContent(bottomLineDiv) === 'true';
+
+  headingBlock.innerHTML = '';
+
+  const eyebrow = document.createElement('p');
+  eyebrow.classList.add('eyebrowtext');
+  eyebrow.textContent = eyebrowText;
+
+  const headerElement = document.createElement(headerTag);
```
Comment (thomsebastin) [inline blocks/heading/heading.js:33]: Could use `dom-utils`.

File: blocks/heading/heading.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const prefixSpan = document.createElement('span');
+  prefixSpan.classList.add('prefix');
```
Comment (thomsebastin) [inline blocks/heading/heading.js:35]: You don't seem to be using this class. Is it needed?

**Review by thomsebastin — CHANGES_REQUESTED** [review]

File: blocks/heading/heading.js (hunk 1 above)
Comment (arjun87743) [reply blocks/heading/heading.js:6]: It was breaking earlier, but its working now with the content thats authored. So removing the check.

**Comment by thomsebastin** [comment]
Approving this PR, few notes for future enhancements:
- Remove empty divs for which content is not authored. This will avoid stale spaces occupied by those elements and improve A11y as well.
- Color the text to be white to be used with dark colored variation. (_This is required and we missed capturing this detail_).
- Explore possibility of different spacing options we can apply. 
@arjun87743

**Review by thomsebastin — APPROVED** [review]

**Pull request merged**